integration-tests/tests/src/tests/observable.ts

```diff
@@ -0,0 +1,848 @@
+
+      const listener = createListenerStub(
+        handle,
+        () => {
+          this.realm.write(() => {
+            collection[0].name = "Bobby";
+          });
+        },
+        () => {
+          collection.removeListener(listener);
+          this.realm.write(() => {
+            collection[0].name = "Charlotte";
+          });
+        },
+      );
+
+      collection.addListener(listener);
+
+      await handle;
+    });
+
+    it("removes all listeners", async function (this: RealmObjectContext<Person>) {
+      const collection = this.object.friends;
+      const handle = createPromiseHandle();
+
+      const listener = createListenerStub(
+        handle,
+        () => {
+          this.realm.write(() => {
+            collection[0].name = "Charles";
+          });
+        },
+        () => {
+          // Remove all listeners and trigger a change
+          collection.removeAllListeners();
+          this.realm.write(() => {
+            collection[0].name = "Charlotte";
+          });
+        },
+      );
+
+      collection.addListener(listener);
+
+      await handle;
+    });
+  });
+
+  describe("Set", () => {
+    type Person = { name: string; friends: Realm.Set<Person> };
+    openRealmBeforeEach({
+      schema: [
+        {
+          name: "Person",
+          properties: {
+            name: "string",
+            friends: { type: "set", objectType: "Person" },
+          },
+        },
+      ],
+    });
+
+    beforeEach(function (this: RealmObjectContext<Person>) {
+      this.object = this.realm.write(() => {
+        const alice = this.realm.create<Person>("Person", { name: "Alice" });
+        const bob = this.realm.create<Person>("Person", { name: "Bob" });
+        alice.friends.add(bob);
+        return alice;
+      });
+    });
+
+    it("is observable", function (this: RealmObjectContext<Person>) {
+      expectObservableMethods(this.object.friends);
+    });
+
+    it("calls listener initially", async function (this: RealmObjectContext<Person>) {
+      const collection = this.object.friends;
+      await expectCollectionNotifications(collection, [
+        {
+          deletions: [],
+          insertions: [],
+          oldModifications: [],
+          newModifications: [],
+        },
+      ]);
+    });
+
+    it("calls listener", async function (this: RealmObjectContext<Person>) {
+      const collection = this.object.friends;
+      const completion = expectCollectionNotifications(collection, [
+        {
+          deletions: [],
+          insertions: [],
+          oldModifications: [],
+          newModifications: [],
+        },
+        {
+          deletions: [],
+          insertions: [0],
+          oldModifications: [],
+          newModifications: [],
+        },
+      ]);
+
+      this.realm.write(() => {
+        // collection["bob"].name = "Bobby";
+        // TODO: It seems we cannot trigger a notification when a property value changes.
```

Comment:
Can we open an issue about it?

Reply:
Either that or we should clarify what the intended behaviour is. I'll follow up on this 👍